Makefile

```diff
@@ -29,29 +29,54 @@ show_bench:
 	@ cat $(BENCH)
 
 .PHONY: dev-clean
+dev-clean: CONTAINER_NAME=apisonator-dev
 dev-clean:
-	-docker kill apisonator-dev
-	-docker rm apisonator-dev
+	-docker kill $(CONTAINER_NAME)
+	-docker rm $(CONTAINER_NAME)
+
+.PHONY: dev-service-clean
+dev-service-clean: CONTAINER_NAME=apisonator-dev-service
+dev-service-clean:
+	-docker kill $(CONTAINER_NAME)
+	-docker rm $(CONTAINER_NAME)
 
 .PHONY: dev-clean-image
 dev-clean-image: IMAGE_NAME?=apisonator-dev
-dev-clean-image: dev-clean
+dev-clean-image: dev-clean dev-service-clean
 	docker rmi $(IMAGE_NAME)
 
 .PHONY: dev
 dev: export IMAGE_NAME?=apisonator-dev
+dev: export CONTAINER_NAME=apisonator-dev
 dev:
 	@docker history -q $(IMAGE_NAME) 2> /dev/null >&2 || $(MAKE) -C $(PROJECT_PATH) -f $(MKFILE_PATH) dev-build
-	@if docker ps --filter name=apisonator-dev | grep -q $(IMAGE_NAME) 2> /dev/null >&2; then \
-		echo "dev container already started" >&2; false ; \
+	@if docker ps --filter name=$(CONTAINER_NAME) --format "{{.Names}}" | grep -q '^$(CONTAINER_NAME)$$' 2> /dev/null >&2; then \
+		echo "$(CONTAINER_NAME) container already started" >&2; false ; \
+	fi
+	@if docker ps -a --filter name=$(CONTAINER_NAME) --format "{{.Names}}" | grep -q '^$(CONTAINER_NAME)$$' 2> /dev/null >&2; then \
+		docker start -ai $(CONTAINER_NAME) ; \
+	else \
+		docker run -ti -h $(CONTAINER_NAME) -v \
+		$(PROJECT_PATH):$$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'cd && pwd')/apisonator:z \
+		-u $$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'id -u'):$$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'id -g') \
+		--name $(CONTAINER_NAME) $(IMAGE_NAME) /bin/bash ; \
+	fi
+
+.PHONY: dev
+dev-service: export IMAGE_NAME?=apisonator-dev
+dev-service: export CONTAINER_NAME=apisonator-dev-service
+dev-service:
+	@docker history -q $(IMAGE_NAME) 2> /dev/null >&2 || $(MAKE) -C $(PROJECT_PATH) -f $(MKFILE_PATH) dev-build
+	@if docker ps --filter name=$(CONTAINER_NAME) --format "{{.Names}}" | grep -q '^$(CONTAINER_NAME)$$' 2> /dev/null >&2; then \
+		echo "$(CONTAINER_NAME) container already started" >&2; false ; \
 	fi
-	@if docker ps -a --filter name=apisonator-dev | grep -q $(IMAGE_NAME) 2> /dev/null >&2; then \
-		docker start -ai apisonator-dev ; \
+	@if docker ps -a --filter name=$(CONTAINER_NAME) --format "{{.Names}}" | grep -w -q '^$(CONTAINER_NAME)$$' 2> /dev/null >&2; then \
+		docker start -ai $(CONTAINER_NAME) ; \
 	else \
 		docker run -ti -h apisonator-dev -v \
 		$(PROJECT_PATH):$$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'cd && pwd')/apisonator:z \
 		-u $$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'id -u'):$$(docker run --rm $(IMAGE_NAME) /bin/bash -c 'id -g') \
-		--name apisonator-dev $(IMAGE_NAME) /bin/bash ; \
+		--name $(CONTAINER_NAME) $(IMAGE_NAME) script/test_external ; \
```

Comment:
forgot to set the container's hostname to `$(CONTAINER_NAME)` a couple of lines above.

Reply:
I did leave the hostname to the hardcoded value because I was not sure if it was some requirement at application level. As we talked the application is not affected when the hostname changes so I will change it.